doc: Rename Ecosystems integration section and template (#10705)
nrfconnect/sdk-nrf: doc/nrf/templates/Integration_template_README.rst

.. _integration_template:

Template: Integration
#####################

.. contents::
   :local:
   :depth: 2

.. note::
   Provide a short introductory description of the product (third-party), indicating the value it brings to |NCS|.
   Refer to the following instructions to create the description:

   * Provide a title that starts with the product name with "integration" as a suffix, for example, "Memfault integration".
   * Place the documentation inside the :file:`nrf/doc/nrf` folder.
   * List the file name with path in the ``.. toctree:`` of the :file:`integrations` RST file.
   * Sections with * are optional and can be left out.
     All other sections are required.
     There could be more sections, which can be added after discussion with the tech writer team.
     However, it should have the mandatory sections from the template in the recommended format at the very least.

.. tip::

>    Describe the product in one or two sentences maximum (full sentences, not sentence fragments), so that the user gets a clear idea of what the product does.

Reviewer: Are we sure 2 sentences are enough? Only Edge Impulse integration is currently matching this requirement.

Side note: do we have tasks for aligning existing integrations with this template?

   This introduction must be concise and clear, highlighting the features of the product that it brings to |NCS|.

> 

Reviewer: Can we have a sample description here, just as we do in line 40?

Partner overview*
*****************

.. note::
   Provide an explanation of the product.

>    This is optional if the product is described above.

Reviewer: ```suggestion
   This is optional if the third-party product is so simple that the introductory two-sentence description is covering all the details.
```
If we're aiming for two sentences in the short desc and we want to convey there what the product does and what it brings to the NCS, we will definitely need this section in my opinion. Two sentences in the short desc for covering everything that should also go into overview is too optimistic assumption.

Prerequisites

> *************

Reviewer: The example you are giving is an example of an integration prerequisite (let's call this A).
The product prerequisite for this example could be something like (totally invented) "create an account on the Memfault page and set up the authentication method Memfault requires" (let's call this B).

I think mentioning A is fine, but mentioning B is outside of our control and we should not track it here. For this reason, I think my suggestion still makes sense.

.. note::
   * Provide prerequisites that must be completed before the integration with the product.

>    * Mention any other product integration that needs to be completed.

Reviewer: What does "any other product integration" mean here? That a third-party product integration might require integrating some other third-party product first? Isn't this just repeating the meaning of line 37?

> Before you start the |NCS| integration with AVSystem's Coiote IoT Device Management, make sure that the following prerequisites are completed:

Reviewer: If we're taking the AVSystem integration as the example here, I would suggest giving it another pass, as it is mixing product with provider in the first two sections (short desc and overview) in its current docs: https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/external_comp/avsystem.html

"AVSystem is a software solution provider that provides a device management platform. "
[...]
"AVSystem provides the following features as a solution partner for Nordic Semiconductor’s cellular devices"

Shouldn't it be "The AVSystem’s Coiote IoT Device Management provides"? Current phrasing could suggest there are more products we're talking here...

Reviewer 2: I think you are correct.

I would also say that if we have different products from the same vendor integrated with NCS then these should be described separately.

Reviewer 3: I shall add the changes for the AVSystem user guide in a separate PR.

.. _integration_template:

Template: Integration
#####################

.. contents::
   :local:
   :depth: 2

.. note::
   Provide a short introductory description of the product or service, indicating the value it brings to |NCS|.
   Refer to the following instructions to create the description:

   * Provide a title that starts with the product or service name with "integration" as a suffix, for example, "Memfault integration".
   * Place the documentation inside the :file:`nrf/doc/nrf` folder.
   * List the file name with path in the ``.. toctree:`` of the :file:`integrations` RST file.
   * Sections with * are optional and can be left out.
     All other sections are required.
     There could be more sections, which can be added after discussion with the tech writer team.
     However, it should have the mandatory sections from the template in the recommended format at the very least.

.. tip::
   Describe the product or service briefly (full sentences, not sentence fragments), so that the user gets a clear idea of what the product or service does.
   This introduction must be concise and clear, highlighting the features of the product or service that it brings to |NCS|.

> `Edge Impulse`_ is a development platform that can be used to enable `embedded machine learning`_ on |NCS| devices.
> You can use this platform to collect data from sensors, train machine learning model, and then deploy it to your Nordic Semiconductor device.

Reviewer: ```suggestion
Edge Impulse is a development platform that can be used to enable embedded machine learning on |NCS| devices.
You can use this platform to collect data from sensors, train machine learning model, and then deploy it to your Nordic Semiconductor device.
```
It is just an example, so I don't think we need to have the valid links here.

> You can use this platform to collect data from sensors, train machine learning model, and then deploy it to your Nordic Semiconductor's device.

Reviewer 3: ```suggestion
You can use this platform to collect data from sensors, train machine learning model, and then deploy it to your Nordic Semiconductor device.
```

*********************

.. note::

>    * Provide a short description of :term:`Development Kit (DK)` and |NCS| interaction with the product and the key elements, for example, connectivity and onboarding.

Reviewer: ```suggestion
   * Provide a short description of structure of the |NCS| integration with the product. Mention the :term:`Development Kit (DK)` used for integration, how it interacts with the product, and what the key elements are, for example connectivity and onboarding.
```

Author: Rephrased the sentence for more clarity.

>    * You can include a diagram showing the interaction.

Reviewer: ```suggestion
   * Include a diagram showing the interaction, if possible.
```

Integration overview
********************

.. note::
   * Explain in more detail how the integration of |NCS| with the product works.

>    * Try to explain the integration in steps.

Reviewer: ```suggestion
   * Explain the integration in steps.
```

>    * Try to explain the integration in steps.

Reviewer: Is the AVSystems numbered header approach the way to follow here?

Also, are we talking here about steps to take for the integration to happen or about overview? Because overview describes things, while steps usually require action.

Author: In the meeting, we discussed and decided to have an integration overview for describing the integration and a subsection on integration steps to explain the steps for integration.
But integration overview would have the same content as integration steps, or the content would mostly fall under solution architecture or other sections. Is it okay to have a single main section called "Integration Steps"?

Please let me know if it's fine or if I missed anything that we discussed during the meeting.

@greg-fer , @richabp , @umapraseeda , @b-gent

Reviewer: Maybe keep the overview as optional (*), so that if there is anything more complex that needs to be explained, we can use this section.

Author: updated the template doc.

     If possible, divide it into sections and provide a brief explanation in the overview with steps.

>      For an example, see the :ref:`ug_integrating_fast_pair` section of the :ref:`ug_bt_fast_pair` documentation.

Reviewer: For lines 64-70: Can we provide information about overlay configuration that is not pushing the user towards command line? The VS Code extension is our primary IDE, and at the very least we should mention it here next to the command line option, and at best we should provide the information about configuration files and leave it to the user to decide which way of applying the configuration to use.

   * Include a diagram showing the interaction, if possible.

Integration overview
********************

.. note::
   * Explain in more detail how the integration of |NCS| with the product or service works.

Integration steps
=================

.. note::
   * Explain the integration in steps.
     If possible, divide it into sections and provide a brief explanation in the overview with steps.
     For an example, see the :ref:`ug_integrating_fast_pair` section of the :ref:`ug_bt_fast_pair` documentation.

.. tip::
   * You can list the configuration that must be enabled for the integration to work (if applicable).
   * You can add information about overlay configuration files and how they are specified in the build system using |VSC| or command line to enable specific features (if applicable).

Some title*
***********

.. note::
   Add optional sections for other technical details about the integration(for example, user-defined configuration options).
   Give suitable titles (sentence style capitalization, thus only the first word capitalized).

>    If there is nothing important to point out, there is no need to include any sections.

Reviewer 3: ```suggestion
   If there is nothing important to point out, you need not include any such section.
```

Applications and samples
************************

.. note::

>    Add the details about applications and samples that use or implement the product or service.

Reviewer 4: ```suggestion
   Add details about applications and samples that use or implement the product or service.
```

> ************************

Reviewer: For lines 72-78: What would be the technical details that need additional section in a step-by-step procedure? I think it's too late to discuss technicalities when we're providing steps for integration... Shouldn't these be included earlier?

Maybe this should be placed under Solution architecture, if the architecture is open to extensions and/or requires some technical details that are not immediately related to the architecture, but still relevant for the product? Although, on second thought, I think this "Some title" section is simply not needed...
